cpp/benchmarks/common/random_distribution_factory.hpp

```diff
@@ -18,17 +18,29 @@
 
 #include "generate_input.hpp"
 
-#include <memory>
-#include <random>
+#include <rmm/device_uvector.hpp>
+
+#include <thrust/execution_policy.h>
+#include <thrust/random.h>
+#include <thrust/random/normal_distribution.h>
+#include <thrust/random/uniform_int_distribution.h>
+#include <thrust/tabulate.h>
 
+#include <cmath>
+#include <memory>
+#include <type_traits>
 /**
  * @brief Generates a normal(binomial) distribution between zero and upper_bound.
  */
 template <typename T, std::enable_if_t<cuda::std::is_integral_v<T>, T>* = nullptr>
 auto make_normal_dist(T upper_bound)
 {
-  using uT = typename std::make_unsigned<T>::type;
-  return std::binomial_distribution<uT>(upper_bound, 0.5);
+  // Provided n is large enough, Normal(μ,σ2) is a good approximation for Binomial(n, p)
+  // where μ = np and σ2 = np (1 - p).
+  using realT        = std::conditional_t<sizeof(T) * 8 <= 23, float, double>;
+  realT const mean   = static_cast<realT>(upper_bound) / 2;             // μ = np, p=0.5
+  realT const stddev = std::sqrt(static_cast<realT>(upper_bound) / 4);  // sqrt(np (1 - p))
```

Comment:
It would also probably help future readers unfamiliar with statistics if the code itself documented the underlying variables (if we do end up sticking to the binomial).
```suggestion
  realT const n = static_cast<realtT>(upper_bound);
  realT const p = 0.5;  // Hardcoded binomial probability of 0.5
  realT const mean   = n * p;
  realT const stddev = std::sqrt(n * p * (1 - p));
```

Reply:
removed.